libs/vkd3d/device.c

```diff
@@ -273,6 +273,49 @@ static unsigned int vkd3d_enable_extensions(const char *extensions[],
     return extension_count;
 }
 
+static bool vkd3d_remove_extension(const char *to_remove, const char *extensions[], uint32_t *extension_count)
+{
+    bool found = false;
+    uint32_t i;
+
+    for (i = 0; i < *extension_count; i++)
+    {
+        if (!found && strcmp(to_remove, extensions[i]) == 0)
+            found = true;
+
+        if (found && i < (*extension_count - 1))
+            extensions[i] = extensions[i+1];
```

Comment:
`i + 1` formatting.

Reply:
This seems a bit inefficient, I'd rather do `if (found) { extensions[i] = extensions[last_index]; extension_count--; }` or something. Ordering is not important.

Reply:
Ah, yes. Since the order does not matter something like this should do:

```c
static bool vkd3d_remove_extension(const char *to_remove, const char *extensions[], uint32_t *extension_count)
{
    uint32_t i;

    for (i = 0; i < *extension_count; i++)
    {
        if (strcmp(to_remove, extensions[i]) != 0)
            continue;

        WARN("Removing %s extension from the array.\n", to_remove);

        if (i < (*extension_count - 1))
            extensions[i] = extensions[*extension_count - 1];

        (*extension_count)--;

        return true;
    }

    return false;
}
```